dotCMS/src/integration-test/java/com/dotcms/publishing/PublisherAPITest.java

```diff
@@ -319,4 +326,293 @@ private FileFilter getNoBundleXMLFileFilter() {
+     * User A - Have Access to the Filter
+     * User B - Have Access to the Filter
+     * User C - Have Access to the Filter
+     * User D - Do not have access to the Filter
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void test_getFiltersByRole_nonCMSAdmin_RoleHierarchy() throws DotDataException {
+        publisherAPI.getFilterDescriptorMap().clear();
+
+        final User userA = new UserDataGen().nextPersisted();
+        final User userB = new UserDataGen().nextPersisted();
+        final User userC = new UserDataGen().nextPersisted();
+        final User userD = new UserDataGen().nextPersisted();
+
+        final Role roleA = new RoleDataGen().nextPersisted();
+        final Role roleB = new RoleDataGen().parent(roleA.getId()).nextPersisted();
+        final Role roleC = new RoleDataGen().parent(roleB.getId()).nextPersisted();
+
+        APILocator.getRoleAPI().addRoleToUser(roleA,userA);
+        APILocator.getRoleAPI().addRoleToUser(roleB,userB);
+        APILocator.getRoleAPI().addRoleToUser(roleC,userC);
+        APILocator.getRoleAPI().addRoleToUser(roleC,userD);
+
+        final Map<String,Object> filtersMap1 =
+                ImmutableMap.of("dependencies",true,"relationships",true,"excludeClasses","Host,Workflow");
+        final FilterDescriptor filterDescriptor1 =
+                new FilterDescriptor("filterTest1.yml","Filter Test Title 1",filtersMap1,false,roleB.getRoleKey()+','+userC.getUserId());
+
+        publisherAPI.addFilterDescriptor(filterDescriptor1);
+
+        //User A
+        List<FilterDescriptor> filterDescriptors = publisherAPI.getFiltersDescriptorsByRole(userA);
+        Logger.info(this,filterDescriptors.toString());
+        Assert.assertFalse(filterDescriptors.isEmpty());
+        Assert.assertTrue(filterDescriptors.contains(filterDescriptor1));
+
+        //User B
+        filterDescriptors = publisherAPI.getFiltersDescriptorsByRole(userB);
+        Logger.info(this,filterDescriptors.toString());
+        Assert.assertFalse(filterDescriptors.isEmpty());
+        Assert.assertTrue(filterDescriptors.contains(filterDescriptor1));
+
+        //User C
+        filterDescriptors = publisherAPI.getFiltersDescriptorsByRole(userC);
+        Logger.info(this,filterDescriptors.toString());
+        Assert.assertFalse(filterDescriptors.isEmpty());
+        Assert.assertTrue(filterDescriptors.contains(filterDescriptor1));
+
+        //User D
+        filterDescriptors = publisherAPI.getFiltersDescriptorsByRole(userD);
+        Logger.info(this,filterDescriptors.toString());
+        Assert.assertTrue(filterDescriptors.isEmpty());
+    }
+
+    private void createFilterDescriptor(final String key,final String title,final boolean defaultFilter,Map<String,Object> filtersMap){
+        if(!UtilMethods.isSet(filtersMap)){
+            final List<Object> listExcludeClasses = new ArrayList();
+            listExcludeClasses.add("User");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "User" appears 5 times in this file; the first occurrence is on line 471](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5458951)